### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by icemac**
Lint the code. 

Add Python 3.9 support to make check-python-versions happy.

### CODE REVIEW

File: src/zope/server/http/publisherhttpserver.py [hunk 1]
```diff
@@ -38,15 +40,17 @@ def __init__(self, request_factory, sub_protocol=None, *args, **kw):
     def _make_application(cls, request_factory, publish=_publish):
         return _make_application(request_factory, publish)
 
+
 def _pmdb_publish(request):
     try:
         return _publish(request, handle_errors=False)
-    except: # pylint:disable=bare-except
+    except:  # noqa: E722 do not use bare 'except'
```
Comment (mgedmin) [inline src/zope/server/http/publisherhttpserver.py:47]: This will make @jamadden sad.

File: src/zope/server/interfaces/ftp.py [hunk 2]
```diff
@@ -22,151 +23,124 @@ class IFTPCommandHandler(Interface):
...
     def cmd_mdtm(args):
         """Show last modification time of file.
 
-           Example output: 213 19960301204320
+        Example output: 213 19960301204320
 
-           Geez, there seems to be a second syntax for this fiel, where one
-           can also set the modification time using:
-           MDTM datestring pathname
+        Geez, there seems to be a second syntax for this fiel, where one
```
Comment (mgedmin) [inline src/zope/server/interfaces/ftp.py:51]: Typo: fiel -> field

File: src/zope/server/interfaces/ftp.py [hunk 3]
```diff
@@ -22,151 +23,124 @@ class IFTPCommandHandler(Interface):
...
 
     def cmd_mkd(args):
-        """Make a directory. Write access required.
-        """
+        """Make a directory. Write access required."""
 
     def cmd_mode(args):
-        """Set file transfer mode.  No read access required. Obselete.
-        """
+        """Set file transfer mode.  No read access required. Obsolete."""
```
Comment (mgedmin) [inline src/zope/server/interfaces/ftp.py:61]: Minor: the number of spaces after the period is inconsistent.

File: src/zope/server/ftp/server.py [hunk 4]
```diff
@@ -31,64 +31,63 @@
...
-    'ERR_DELETE_DIR'   : '550 Error removing directory: %s',
-    'ERR_OPEN_READ'    : '553 Could not open file for reading: %s',
-    'ERR_OPEN_WRITE'   : '553 Could not open file for writing: %s',
-    'ERR_IO'           : '553 I/O Error: %s',
-    'ERR_RENAME'       : '560 Could not rename "%s" to "%s": %s',
-    'ERR_RNFR_SOURCE'  : '560 No source filename specify. Call RNFR first.',
-    }
+    'OPEN_DATA_CONN': '150 Opening %s mode data connection for file list',
+    'OPEN_CONN': '150 Opening %s connection for %s',
+    'SUCCESS_200': '200 %s command successful.',
```
Comment (mgedmin) [inline src/zope/server/ftp/server.py:36]: This makes me sad, I'd rather we disabled this pycodestyle check in a setup.cfg or a .flake8

**Review by mgedmin — APPROVED** [review]
LGTM

There are some minor typos etc. that were there in the old code as well, but we might fix them now that we noticed them.

File: src/zope/server/http/publisherhttpserver.py (hunk 1 above)
Comment (jamadden) [reply src/zope/server/http/publisherhttpserver.py:47]: It does make me sad. This should also work, without losing anything:
```suggestion
    except:  # noqa: E722 pylint:disable=bare-except
```

Comment (ale-rt) [reply src/zope/server/http/publisherhttpserver.py:47]: is `except Exception:` an option or is this code expect to catch exceptions that do not inherit from Exception?

Comment (ale-rt) [reply src/zope/server/http/publisherhttpserver.py:47]: Well this is the exact reason why a bare `except:` is not considered good... you do not know what to expect ;)

File: src/zope/server/ftp/server.py (hunk 4 above)
Comment (icemac) [reply src/zope/server/ftp/server.py:36]: I reverted the change and are now ignoring `E203 whitespace before ':'` for this file.

**Pull request merged**